### PR TITLE
Correctly detect if KeyManagerFactory is supported by OpenSSL even wh…

### PR DESCRIPTION
**Pull request by normanmaurer**
…en sun.security.x509.* can not be accessed and bouncycastle is not on the classpath.

Motivation:

OpenSsl used SelfSignedCertificate in its static init block to detect if KeyManagerFactory is supported. Unfortunally this only works when either sun.security.x509.* can be accessed or bouncycastle is on the classpath.
We should not depend on either of it.

This came up in https://github.com/netty/netty-tcnative/issues/404#issuecomment-431551890.

Modifications:

Just directly use the bytes to generate the X509Certificate and so not depend on sun.security.x509.* / bouncycastle.

Result:

Correctly be able to detect if KeyManagerFactory can be supported in all cases.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@ejona86 @carl-mastrangelo @trustin let me know if I should use a different hostname for the self-signed cert bytes. That said it really not matters at all.

File: handler/src/main/java/io/netty/handler/ssl/OpenSsl.java [hunk 1]
```diff
@@ -172,8 +173,40 @@
                             logger.debug("Hostname Verification not supported.");
                         }
                         try {
-                            cert = new SelfSignedCertificate();
-                            certBio = ReferenceCountedOpenSslContext.toBIO(ByteBufAllocator.DEFAULT, cert.cert());
+                            // Bytes of Self-Signed certificate for netty.io
+                            byte[] certBytes =
+                                    { 48, -126, 1, -92, 48, -126, 1, 13, -96, 3, 2, 1, 2, 2, 9, 0, -9, 61,
```
Comment (trustin) [inline handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: nit: Curly brace should be at the end of the previous line.

File: handler/src/main/java/io/netty/handler/ssl/OpenSsl.java [hunk 2]
```diff
@@ -172,8 +173,40 @@
                             logger.debug("Hostname Verification not supported.");
                         }
                         try {
-                            cert = new SelfSignedCertificate();
-                            certBio = ReferenceCountedOpenSslContext.toBIO(ByteBufAllocator.DEFAULT, cert.cert());
+                            // Bytes of Self-Signed certificate for netty.io
```
Comment (trustin) [inline handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: nit: Self-Signed -> self-signed

Comment (normanmaurer) [reply handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: done

**Pull request merged**